### PR TITLE
[#18][Part1] [UI] Survey Details screen

### LINKED ISSUE #18: [UI] As a user, I can see the survey details when taking a survey
## Why

App must display the survey questions screen when taking a survey

## Acceptance Criteria

- Must display `Survey.coverImageUrl` as background
- Must display question index and total question count
  - do not include intro question
- If first question, 
  - show `Take Survey` button
  - show back button 
- if last question
  - show `Submit Survey` button
  - show close button
- if not first or last question
  - display Floating Action Button with right chevron icon
  - show close button

## Resources

- [Design](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=34%3A868)

### PR DESCRIPTION
**Pull request by lydiasama**
- #18 

## What happened 👀

- [x] [Part 1]: Set up AppNavigation. 

- [ ] [Part 2]: Implement UI for Survey Details screen

## Insight 📝

- When using Stream to check if user logged in before navigating to the Home screen, there is an issue that it does not navigate to home as expected.
- I found that we can use [Redirect feature](https://pub.dev/documentation/go_router/latest/topics/Redirection-topic.html) from Go_router to consider navigating to specify path at the top-level. So no longer using `VerifyLoggedInUseCase`. 👉🏻  Checking the token directly from `main.dart` as we don't use the use case in main.
- Implement to navigate to `HomeScreen` after user has logged in.
- Updated unit test.

## Proof Of Work 📹

https://user-images.githubusercontent.com/12026942/233967267-bdb71996-b4d3-48ff-b7bc-c706c8486cb9.mp4

### CODE REVIEW

File: lib/screens/login/login_screen.dart [hunk 1]
```diff
@@ -50,18 +47,11 @@ class _LoginScreenState extends ConsumerState<LoginScreen> {
...
-    if (isLoggedIn) {
-      showToast('Logged In. Will navigate to Home Screen');
-      // TODO _navigateToHome()
-      return Container();
-    }
-
     ref.listen<BaseViewModelState>(loginViewModelProvider, (_, state) {
       state.maybeWhen(
         success: () {
           showToast('Login Success');
```
Comment (ryan-conway) [inline lib/screens/login/login_screen.dart:53]: [Optional] Shall we remove this toast now? 😄

**Review by ryan-conway — APPROVED** [review]
👌🏻

**Review by markgravity — APPROVED** [review]

File: lib/screens/login/login_screen.dart (hunk 1 above)
Comment (lydiasama) [reply lib/screens/login/login_screen.dart:53]: @ryan-conway Let me remove it at part2. 😄

Comment (lydiasama) [reply lib/screens/login/login_screen.dart:53]: @ryan-conway Removed at 838f9a51 🙏🏻

**Pull request merged**